editor/src/clj/editor/atlas.clj

```diff
@@ -914,6 +926,21 @@
                                                               child-build-errors
                                                               own-build-errors))))
 
+(defn- get-node-by-type [child-node-type]
+  (fn get-child-node-by-type [node evaluation-context]
+    (let [basis (:basis evaluation-context)]
+      (filterv #(= child-node-type (g/node-type* basis %)) (g/node-value node :nodes evaluation-context)))))
```

Comment:
Name suggests it returns a single node, but it returns a vector of nodes.

Instead of matching by node-type like this, it might be more efficient to use `g/explicit-arcs-by-target` on the relevant input to obtain the list of nodes. 🤔